packages/twenty-server/src/workspace/calendar/services/google-calendar-full-sync.service.ts

```diff
@@ -0,0 +1,270 @@
+        endTime - startTime
+      }ms.`,
+    );
+
+    // TODO: In V2, we will also import deleted events by doing batch GET queries on the canceled events
+    // The canceled events start and end are not accessible in the list query
+    const formattedEvents = events
+      .filter((event) => event.status !== 'cancelled')
+      .map((event) => formatGoogleCalendarEvent(event));
+
+    // TODO: When we will be able to add unicity contraint on iCalUID, we will do a INSERT ON CONFLICT DO UPDATE
+
+    const existingEventExternalIds =
+      existingCalendarChannelEventAssociations.map(
+        (association) => association.eventExternalId,
+      );
+
+    const eventsToSave = formattedEvents.filter(
+      (event) => !existingEventExternalIds.includes(event.externalId),
+    );
+
+    const eventsToUpdate = formattedEvents.filter((event) =>
+      existingEventExternalIds.includes(event.externalId),
+    );
+
+    const calendarChannelEventAssociationsToSave = eventsToSave.map(
+      (event) => ({
+        calendarEventId: event.id,
+        eventExternalId: event.externalId,
+        calendarChannelId,
+      }),
+    );
+
+    const attendeesToSave = eventsToSave.flatMap((event) => event.attendees);
+
+    const attendeesToUpdate = eventsToUpdate.flatMap(
+      (event) => event.attendees,
+    );
+
+    const iCalUIDCalendarEventIdMap =
+      await this.calendarEventService.getICalUIDCalendarEventIdMap(
+        eventsToUpdate.map((event) => event.iCalUID),
+        workspaceId,
+      );
+
+    if (events.length > 0) {
+      const dataSourceMetadata =
+        await this.workspaceDataSourceService.connectToWorkspaceDataSource(
+          workspaceId,
+        );
+
+      dataSourceMetadata?.transaction(async (transactionManager) => {
+        this.calendarEventService.saveCalendarEvents(
+          eventsToSave,
+          workspaceId,
+          transactionManager,
+        );
+
+        this.calendarEventService.updateCalendarEvents(
+          eventsToUpdate,
+          workspaceId,
+          transactionManager,
+        );
+
+        this.calendarChannelEventAssociationService.saveCalendarChannelEventAssociations(
+          calendarChannelEventAssociationsToSave,
+          workspaceId,
+          transactionManager,
+        );
+
+        this.calendarEventAttendeesService.saveCalendarEventAttendees(
+          attendeesToSave,
+          workspaceId,
+          transactionManager,
+        );
+
+        this.calendarEventAttendeesService.updateCalendarEventAttendees(
+          attendeesToUpdate,
+          iCalUIDCalendarEventIdMap,
+          workspaceId,
+          transactionManager,
+        );
+      });
+    } else {
+      this.logger.log(
+        `google calendar full-sync for workspace ${workspaceId} and account ${connectedAccountId} done with nothing to import.`,
+      );
+    }
```

Comment:
Feel like if events array is empty we're doing nothing, so I rather put the `if (events.length === 0)` first and then return after the log

Reply:
For now yes, but after we will update the sync cursor (commented code below: this.calendarChannelService.updateSyncCursor)